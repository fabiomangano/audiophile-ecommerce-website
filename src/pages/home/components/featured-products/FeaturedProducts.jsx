import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useMediaQuery } from 'react-responsive';
import { MediumOrLargeScreen } from '../../../../mediaQueries';
import Marginer from '../../../../components/marginer/Marginer';
import Button from '../../../../components/button/Button';
import endpoints from '../../../../config/endpoint.json';
import style from './FeaturedProducts.module.scss';

const FeaturedProducts = ({ featuredProducts, fetchFeaturedProducts }) => {
  const isMediumOrLargeScreen = useMediaQuery(MediumOrLargeScreen);

  useEffect(() => {
    fetchFeaturedProducts(endpoints['featured-products']);
  }, []);

  if (!featuredProducts) {
    return (
      <>
        <div className={style.main} />
        <Marginer direction="horizontal" margin={2.2} />
        <div className={style.flat} />
        <Marginer direction="horizontal" margin={2.2} />
        <div className={style.splitted}>
          <div className={style.splitted_top} />
          <div className={style.splitted_bottom} />
        </div>
      </>
    );
  }

  return (
    <>
      <div className={style.main}>
        <Marginer direction="horizontal" margin={3.5} />
        <img
          src={featuredProducts[0].url}
          alt="Product"
          className={style.main_image}
        />
        <Marginer direction="horizontal" margin={3.5} />
        <div className={style.main_content}>
          <h1>{featuredProducts[0].name}</h1>
          <Marginer direction="horizontal" margin={3.5} />
          <p>{featuredProducts[0].description}</p>
          <Marginer direction="horizontal" margin={3.5} />
          <Button variant="secondary">SEE PRODUCT</Button>
        </div>
      </div>
      <Marginer direction="horizontal" margin={2.2} />
      <div className={style.flat}>
        <div className={style.flat_content}>
          <h3>{featuredProducts[1].name}</h3>
          <Marginer direction="horizontal" margin={2} />
          <Button variant="outline">SEE PRODUCT</Button>
        </div>
      </div>
      <Marginer direction="horizontal" margin={2.2} />
      <div className={style.splitted}>
        <div className={style.splitted_top} />
        <Marginer
          direction={isMediumOrLargeScreen ? 'vertical' : 'horizontal'}
          margin={1.5}
        />
        <div className={style.splitted_bottom}>
          <h3>{featuredProducts[1].name}</h3>
          <Marginer direction="horizontal" margin={1.5} />
          <Button variant="outline">SEE PRODUCT</Button>
        </div>
      </div>
    </>
  );
};

FeaturedProducts.propTypes = {
  featuredProducts: PropTypes.arrayOf(
    PropTypes.objectOf(
      PropTypes.shape({
        id: PropTypes.String,
        url: PropTypes.String,
        name: PropTypes.String,
        description: PropTypes.String,
      }),
    ),
  ).isRequired,
  fetchFeaturedProducts: PropTypes.func.isRequired,
};

export default FeaturedProducts;
