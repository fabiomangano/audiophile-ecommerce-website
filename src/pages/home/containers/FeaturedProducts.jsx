import { connect } from 'react-redux';
import { fetchFeaturedProducts } from '../../../redux/actions/feautured-products/featuredProducts';
import FeaturedProducts from '../components/featured-products/FeaturedProducts';

const mapStateToProps = (state) => {
  return {
    featuredProducts: state.featuredProducts.data,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    fetchFeaturedProducts: (url) => dispatch(fetchFeaturedProducts(url)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(FeaturedProducts);
