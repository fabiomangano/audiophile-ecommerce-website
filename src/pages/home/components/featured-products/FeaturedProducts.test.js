import React from 'react';
import { create } from 'react-test-renderer';
import FeaturedProducts from './FeaturedProducts';

describe('FeauturedProducts', () => {
  it('should renders correctly', () => {
    const tree = create(<FeaturedProducts />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
