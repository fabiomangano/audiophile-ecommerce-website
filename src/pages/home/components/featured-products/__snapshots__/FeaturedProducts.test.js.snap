// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeauturedProducts should renders correctly 1`] = `
Array [
  <div
    className="main"
  />,
  <span
    style={
      Object {
        "display": "flex",
        "height": "2.2rem",
      }
    }
  />,
  <div
    className="flat"
  />,
  <span
    style={
      Object {
        "display": "flex",
        "height": "2.2rem",
      }
    }
  />,
  <div
    className="splitted"
  >
    <div
      className="splitted_top"
    />
    <div
      className="splitted_bottom"
    />
  </div>,
]
`;
